src/main/java/org/asamk/signal/Manager.java

```diff
@@ -775,6 +775,59 @@ public void sendEndSessionMessage(List<String> recipients) throws IOException, E
         sendMessage(messageBuilder, recipients);
     }
 
+    @Override
+    public String getContactName(String number) {
+        ContactInfo contact = contactStore.getContact(number);
+        if(contact == null) {
+            return number;
+        } else {
+            return contact.name;
+        }
+    }
+
+    @Override
+    public void setContactName(String number, String name) {
+        ContactInfo contact = contactStore.getContact(number);
+        if(contact == null) {
+            contact = new ContactInfo();
+            contact.number = number;
+            System.out.println("Add contact " + number + " named " + name);
+        } else {
+            System.out.println("Updating contact " + number + " name " + contact.name + " -> " + name);
+        }
+        contact.name = name;
+        contactStore.updateContact(contact);
+        save();
+    }
+
+    @Override
+    public String getGroupName(byte[] groupId) {
+        return getGroup(groupId).name;
+    }
+
+    @Override
+    public List<String> getGroupMembers(byte[] groupId) {
+        return new ArrayList<String>(getGroup(groupId).members);
+    }
+
+    @Override
+    public void updateGroup(byte[] groupId, String name, List<String> members, String avatar) throws IOException, EncapsulatedExceptions, GroupNotFoundException, AttachmentInvalidException {
+        String optName = null;
+        Collection<String> optMembers = null;
+        String optAvatar = null;
+        if(!name.isEmpty()) {
+            optName = name;
+        }
+        if(members.size() > 0) {
+            optMembers = members;
+        }
+        if(!avatar.isEmpty()) {
+            optAvatar = avatar;
+        }
+        System.err.println("sendUpdateGroupMessage(" + Base64.encodeBytes(groupId) + ", " + optName + ", " + optMembers + ", " + optAvatar + ");");
```

Comment:
oops, didn't realize I left this in there at all, I was just using it for debugging. Would you rather I simply remove it?